packages/collaboration-manager/src/Operation.spec.ts

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable @typescript-eslint/no-magic-numbers */
+import type { DataKey, DocumentIndex } from '@editorjs/model';
+import { IndexBuilder } from '@editorjs/model';
+import { Operation, OperationType } from './Operation.js';
+
+describe('Operation', () => {
+  const createOperation = (type: OperationType, startIndex: number, value: string): Operation => {
+    return new Operation(
+      type,
+      new IndexBuilder()
+        .addBlockIndex(0)
+        .addDataKey('text' as DataKey)
+        .addTextRange([startIndex, startIndex])
+        .build(),
+      {
+        prevValue: type === OperationType.Delete ? value : '',
+        newValue: type === OperationType.Insert ? value : '',
+      }
+    );
+  };
+
+  describe('Insert vs Insert', () => {
+    test('Should not change a received operation if it is before a local one', () => {
+      const receivedOp = createOperation(OperationType.Insert, 0, 'abc');
+      const localOp = createOperation(OperationType.Insert, 3, 'def');
+      const transformedOp = receivedOp.transform(localOp);
+
+      expect(transformedOp).toEqual(receivedOp);
+    });
+
+    test('Should adjust an index for a received operation if it is after a local one', () => {
+      const receivedOp = createOperation(OperationType.Insert, 3, 'def');
+      const localOp = createOperation(OperationType.Insert, 0, 'abc');
+      const transformedOp = receivedOp.transform(localOp);
+
+      expect(transformedOp.index.textRange).toEqual([6, 6]);
+    });
+
+    test('Should not change a received operation if it is at the same position as a local one', () => {
```

Comment:
If position is the same, the index should change by the length of text in received operation